sos/plugins/cc-runtime.py

```diff
@@ -0,0 +1,30 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from sos.plugins import Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin, \
+        SuSEPlugin
+
+
+class ClearContainers(Plugin, RedHatPlugin, DebianPlugin, UbuntuPlugin,
+                      SuSEPlugin):
+    """Intel(R) Clear Containers configuration
+    """
+
+    plugin_name = 'clear-containers'
+    profiles = ('system', 'virt')
+
+    packages = ('cc-runtime',)
+
+    def setup(self):
+        self.add_cmd_output('cc-collect-data.sh')
```

Comment:
Looking at this script in the runtime repository it duplicates functionality that's already present in other plugins (Kubernetes, Docker, KVM, package listings that are provided by the rpm and dpkg plugins). It's also capturing logs from components in a way that is not in keeping with what sos's users expect: they are not subject to the global `--log-size` and `--all-logs` switches and will not appear at the conventional locations in the generated report.

There's nothing very complex going on in this script and I think it would be better to implement this natively in Python as the other plugins in the tree do today. The one aspect that's potentially useful is the log grepping: this could be done in the plugin itself, or, if the script allows a method to disable all the other unnecessary collection we could carry on calling out to it for that.

Reply:
Thanks for the prompt feedback @bmr-cymru. I'll rework this PR as suggested...